### ISSUE TITLE
rpc: need wait at least 10 + 1 maxItems in waitscan brings delay ux

### ISSUE THREAD

**Issue by mmsqe** [tendermint/tendermint#9369]
**Tendermint version** : current master

**What happened**: rpc: need wait at least 10 + 1 with [hardcoded](https://github.com/tendermint/tendermint/blob/85a584dd2ee3437b117d9f5e894eac70bf09aeef/internal/rpc/core/events.go#L159) maxItems in waitscan brings delay ux

**What you expected to happen**: nice to have more realtime response

**Comment by thanethomson** [comment]
> Tendermint version : current master

The `master` branch code is no longer supported. Our default branch is `main` now, which is based on `v0.34.x`.

Is this still an issue on the `main` branch? If so, please give us a more detailed description of the problem. It's not totally clear to me.

**Comment by mmsqe** [comment]
> > Tendermint version : current master
> 
> The `master` branch code is no longer supported. Our default branch is `main` now, which is based on `v0.34.x`.
> 
> Is this still an issue on the `main` branch? If so, please give us a more detailed description of the problem. It's not totally clear to me.

Got it, it's related to https://github.com/tendermint/tendermint/issues/9305. You can see from [here](https://github.com/tendermint/tendermint/blob/85a584dd2ee3437b117d9f5e894eac70bf09aeef/internal/rpc/core/events.go#L197), a `WaitScan` only returns when no. of items exceed `maxItems`. Having minimum of 10 `maxItems` stated [here](https://github.com/tendermint/tendermint/blob/85a584dd2ee3437b117d9f5e894eac70bf09aeef/internal/rpc/core/events.go#L159), we cannot stop cursor and return once 1 item is received, and need to wait indefinitely until 11 items received.

**Comment by creachadair** [comment]
> Got it, it's related to https://github.com/tendermint/tendermint/issues/9305. You can see from [here](https://github.com/tendermint/tendermint/blob/85a584dd2ee3437b117d9f5e894eac70bf09aeef/internal/rpc/core/events.go#L197), a `WaitScan` only returns when no. of items exceed `maxItems`. Having minimum of 10 `maxItems` stated [here](https://github.com/tendermint/tendermint/blob/85a584dd2ee3437b117d9f5e894eac70bf09aeef/internal/rpc/core/events.go#L159), we cannot stop cursor and return once 1 item is received, and need to wait indefinitely until 11 items received.

That isn't quite correct. The call only waits up to the polling interval, which by default is one second. Once that expires, regardless how many results there are, the call reports what it has.

**Comment by mmsqe** [comment]
> That isn't quite correct. The call only waits up to the polling interval, which by default is one second. Once that expires, regardless how many results there are, the call reports what it has.
yes, but will be too much requests with 1s maxWait.

**Comment by mmsqe** [comment]
> That isn't quite correct. The call only waits up to the polling interval, which by default is one second. Once that expires, regardless how many results there are, the call reports what it has.

yes, but will be too much requests with 1s maxWait, more like short poll then.
